addons/stock/models/stock_warehouse.py

```diff
@@ -93,6 +93,26 @@ def _default_name(self):
         ('warehouse_code_uniq', 'unique(code, company_id)', 'The short name of the warehouse must be unique per company!'),
     ]
 
+    def _init_column(self, column_name):
+        """ Ensure the default warehouse_id is correctly assigned
+
+        At column initialization, the ir.model.fields for res.users.property_warehouse_id isn't created,
+        which means trying to read the property field to get the default value will crash.
+        We therefore enforce the default here, without going through
+        the default function on the warehouse_id field.
+        """
+        if column_name != "warehouse_id":
+            return super(Warehouse, self)._init_column(column_name)
+        field = self._fields[column_name]
+        default = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+        value = field.convert_to_write(default, self)
+        value = field.convert_to_column(value, self)
+        if value is not None:
+            _logger.debug("Table '%s': setting default value of new column %s to %r",
+                self._table, column_name, value)
+            query = f'UPDATE "{self._table}" SET "{column_name}" = %s WHERE "{column_name}" IS NULL'
+            self._cr.execute(query, (value,))
+
```

Comment:
Sorry, at the time I was moving every reference of property_warehouse_id from sale_stock to stock, and since it didn't crash, I left it as is. I'll move it back.
As for your first remark at the top, my thought was since it was your code I was moving, I should ask your opinion on it.